wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java

```diff
@@ -643,22 +643,24 @@ private static DataSection parseDataSection(
         var dataSegments = new DataSegment[(int) dataSegmentCount];
 
         for (var i = 0; i < dataSegmentCount; i++) {
-            var memidx = readVarUInt32(buffer);
-            // if memory idx == 0, then this is an active datasegment
-            // if memory idx == 1, then this is a passive datasegment
-            // TODO won't this break once we support multi-memory?
-            if (memidx == 0) {
+            var mode = readVarUInt32(buffer);
```

Comment:
Nice one!
This covers all the cases now:
https://webassembly.github.io/spec/core/binary/modules.html#data-section

🎉  do you want to change the explicit throw of the exception to an assertion?

Reply:
For something like this I'd prefer the throw - if someone tries to parse a newer version of a binary (or an incorrectly version labeled binary, or a malformed binary) then it's conceivable that parsing might make it this far.

Reply:
just a personal preference, not fussy on this 🤷